tests/integration/test_reactive.py

```diff
@@ -4,137 +4,227 @@
     _add_to_queue(
         json.dumps(json.loads(job.json()) | {"ignored_noise": "foobar"}),
         mongodb_uri,
-        app_for_reactive.name,
+        app.name,
     )
 
-    await reconcile(app_for_reactive, app_for_reactive.model)
+    # This reconcile call is to check that we are not killing machines that are under
+    # construction in a subsequent reconciliation.
+    await reconcile(app, app.model)
 
-    await wait_for_completion(run, conclusion="success")
-    _assert_queue_is_empty(mongodb_uri, app_for_reactive.name)
+    try:
+        await wait_for_completion(run, conclusion="success")
+    except TimeoutError:
+        assert False, (
+            "Job did not complete successfully, check the reactive log using tmate,"
+            " it might be due to infrastructure issues"
+        )
 
-    # Call reconcile to enable cleanup of the runner
-    await reconcile(app_for_reactive, app_for_reactive.model)
+    _assert_queue_is_empty(mongodb_uri, app.name)
 
 
+@pytest.mark.abort_on_fail
 async def test_reactive_mode_does_not_consume_jobs_with_unsupported_labels(
     ops_test: OpsTest,
-    app_for_reactive: Application,
+    app: Application,
     github_repository: Repository,
     test_github_branch: Branch,
 ):
     """
-    arrange: A charm integrated with mongodb and an unsupported label is added to the queue.
+    arrange: Place a message with an unsupported label in the queue and dispatch a workflow.
     act: Call reconcile.
-    assert: No runner is spawned and the message is requeued.
+    assert: No runner is spawned and the message is not requeued.
     """
-    mongodb_uri = await _get_mongodb_uri(ops_test, app_for_reactive)
+    mongodb_uri = await _get_mongodb_uri(ops_test, app)
 
     run = await dispatch_workflow(
-        app=app_for_reactive,
+        app=app,
         branch=test_github_branch,
         github_repository=github_repository,
         conclusion="success",  # this is ignored currently if wait=False kwarg is used
         workflow_id_or_name=DISPATCH_TEST_WORKFLOW_FILENAME,
         wait=False,
     )
-    jobs = list(run.jobs())
-    assert len(jobs) == 1, "Expected 1 job to be created"
-    job = jobs[0]
-    job_url = job.url
-    job = JobDetails(labels={"not supported label"}, url=job_url)
+    job = _create_job_details(run=run, labels={"not supported label"})
     _add_to_queue(
         job.json(),
         mongodb_uri,
-        app_for_reactive.name,
+        app.name,
     )
 
-    await reconcile(app_for_reactive, app_for_reactive.model)
+    # wait for queue being empty, there could be a race condition where it takes some
+    # time for the job message to be consumed and the queue to be empty
+    try:
+        await wait_for(lambda: _get_queue_size(mongodb_uri, app.name) == 0)
+        run.update()
+        assert run.status == "queued"
+    finally:
+        run.cancel()  # cancel the run to avoid a queued run in GitHub actions page
```

Comment:
This may be unnecessary, but I don't like seeing queued jobs on the actions page.